File: OnlineRadyoJava/app/src/main/java/com/example/onlineradyojava/ui/RoomAdapter.java
package com.example.onlineradyojava.ui;



import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.onlineradyojava.data.RadioRoom;
import com.example.onlineradyojava.databinding.ItemRoomBinding;

import java.util.ArrayList;
import java.util.List;

public class RoomAdapter extends RecyclerView.Adapter<RoomAdapter.RoomViewHolder> {

    private List<RadioRoom> roomList = new ArrayList<>();
    private OnRoomClickListener listener;

    public RoomAdapter(OnRoomClickListener listener) {
        this.listener = listener;
    }

    public void updateRooms(List<RadioRoom> newRooms) {
        roomList.clear();
        roomList.addAll(newRooms);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public RoomViewHolder onCreateViewHolder(
            @NonNull ViewGroup parent,
            int viewType
    ) {
        ItemRoomBinding binding = ItemRoomBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );

        return new RoomViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(
            @NonNull RoomViewHolder holder,
            int position
    ) {
        holder.bind(roomList.get(position));
    }

    @Override
    public int getItemCount() {
        return roomList.size();
    }

    class RoomViewHolder extends RecyclerView.ViewHolder {

        private ItemRoomBinding binding;

        public RoomViewHolder(ItemRoomBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(RadioRoom room) {
            binding.tvRoomName.setText(room.getRoomName());

            if (room.getCurrentMusic() == null || room.getCurrentMusic().isEmpty()) {
                binding.tvCurrentMusic.setText("Şu an: Müzik yok");
            } else {
                binding.tvCurrentMusic.setText("Şu an: " + room.getCurrentMusic());
            }

            binding.tvListenerCount.setText("Dinleyici: " + room.getListenerCount());

            binding.getRoot().setOnClickListener(v -> {
                if (listener != null) {
                    listener.onRoomClick(room);
                }
            });
        }
    }

    public interface OnRoomClickListener {
        void onRoomClick(RadioRoom room);
    }
}

File: OnlineRadyoKotlin/app/src/main/java/com/example/onlineradyokotlin/ui/RoomAdapter.kt
package com.example.onlineradyokotlin.ui


import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.onlineradyokotlin.data.RadioRoom
import com.example.onlineradyokotlin.databinding.ItemRoomBinding

class RoomAdapter(
    private val onRoomClicked: (RadioRoom) -> Unit
) : RecyclerView.Adapter<RoomAdapter.RoomViewHolder>() {

    private val roomList = mutableListOf<RadioRoom>()

    fun updateRooms(newRooms: List<RadioRoom>) {
        roomList.clear()
        roomList.addAll(newRooms)
        notifyDataSetChanged()
    }

    inner class RoomViewHolder(
        private val binding: ItemRoomBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(room: RadioRoom) {
            binding.tvRoomName.text = room.roomName

            binding.tvCurrentMusic.text =
                if (room.currentMusic.isNullOrBlank()) {
                    "Şu an: Müzik yok"
                } else {
                    "Şu an: ${room.currentMusic}"
                }

            binding.tvListenerCount.text = "Dinleyici: ${room.listenerCount}"

            binding.root.setOnClickListener {
                onRoomClicked(room)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RoomViewHolder {
        val binding = ItemRoomBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return RoomViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RoomViewHolder, position: Int) {
        holder.bind(roomList[position])
    }

    override fun getItemCount(): Int = roomList.size
}